applicationset/controllers/applicationset_controller.go

```diff
@@ -867,15 +867,15 @@ func (r *ApplicationSetReconciler) performProgressiveSyncs(ctx context.Context,
 		logCtx.Infof("step %v: %+v", i+1, step)
 	}
 
-	appSyncMap := r.buildAppSyncMap(appset, appDependencyList, appMap)
+	appSyncMap := r.buildAppSyncMap(*appset, appDependencyList, appMap)
```

Comment:
I don't think a pointer is required here either?

Reply:
The `buildAppSyncMap` function expects an `argov1alpha1.ApplicationSet` value, not a pointer. Therefore, we need to dereference the `appset ` here to match the expected type